### ISSUE TITLE
CustomNonbondedForce with InteractionGroup

### ISSUE THREAD

**Issue by isjoung** [pandegroup/openmm#1431]
```python
from simtk import openmm
force = openmm.CustomNonbondedForce('r')
force.addInteractionGroup([0,1,2], [4,5,6])
```

Traceback (most recent call last):
  File "t.py", line 3, in <module>
    force.addInteractionGroup([0,1,2], [4,5,6])
  File "/net/isjoung/.local/lib/python2.7/site-packages/simtk/openmm/openmm.py", line 10208, in addInteractionGroup
    args[0] = list(args[0])
TypeError: 'tuple' object does not support item assignment

Is this a bug or is my syntax wrong?

**Comment by Lnaden** [comment]
I can recreate this bug by building from source. 

The openmm.py function `addInteractionGroup` accepts `*args` which bundles all arguments as a tuple object. That line is trying to turn args[0] into a list, then reassign it to an element of args which is an immutable tuple, throwing the error.

I think this was introduced in #1264 from what I can tell. I'm not the most familiar with SWIG though, so I don't have a proper fix. If you need a quick workaround, you can add the line `args = list(args)` in the openmm.py file right above the line that threw the error.

**Comment by peastman** [comment]
I can't reproduce this.  What version of SWIG are you using?  If I recall, some older versions didn't handle named arguments correctly and instead packed everythng into an `args` tuple (as you described), even when you've told it not to.

I'm using SWIG 3.0.8, which I think is the latest version.

**Comment by rmcgibbo** [comment]
We have slightly different code paths for swig<=3.0.4 and later versions of
swigs, so this is probably an issue in the former codepath.

On Wed, Mar 23, 2016 at 2:51 PM, peastman <notifications@github.com> wrote:

> I can't reproduce this. What version of SWIG are you using? If I recall,
> some older versions didn't handle named arguments correctly and instead
> packed everythng into an args tuple (as you described), even when you've
> told it not to.
>
> I'm using SWIG 3.0.8, which I think is the latest version.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pandegroup/openmm/issues/1431#issuecomment-200492823>
>



-- 
-Robert

**Comment by peastman** [comment]
Perhaps we should just require SWIG 3.0.5 or later?  Are there any situations where it isn't practical for people to use a recent version?

**Comment by swails** [comment]
I don't always use a version later than 3.0.5, but I'm also not against making this requirement.  It's not too hard to install SWIG 3.0.4 and up, and people that install from source (rather than binary) know they are taking on more responsibility anyway.

**Comment by Lnaden** [comment]
I'm still back on SWIG 1.3.40 which is current version pushed through CentOS 6's yum repos. Let me pull down a newer version and check that.

**Comment by Lnaden** [comment]
Yup, newer SIG version fixed it (tested on 3.0.8). SWIG 3.0.5 should probably be the minimum version if we want to enforce a minimum as the SWIG site recommends Python users install 3.0.5 over 3.0.4.

**Comment by isjoung** [comment]
Newer SWIG fixed the problem. I was also on 1.3.40, previously. Thank you.
